data_model/tests/evaluates_to_serde_fail/deser_evaluates_to_fails.rs

```diff
@@ -0,0 +1,17 @@
+use serde::{Serialize, Deserialize};
+
+use iroha_data_model::prelude::*;
+
+#[derive(Serialize, Deserialize)]
+struct MyTestStructEvsTo {}
+
+fn main() {
+    // Generic type does not matter in this context
+    let evals_to = EvaluatesTo::<Value>::new_unchecked(
+        Value::Bool(true),
+    );
```

Comment:
```suggestion
    let evals_to = EvaluatesTo::<Name>::new_unchecked(
        Value::Bool(true),
    );
```

and what if we have a test like this? will this deserialize successfully?

Reply:
this check needs to be done at runtime. You cannot make it a compile time check

Reply:
what you need to do is inspect `Expression` and verify that the contained value is indeed as declared in the type

Reply:
It will not.
But it will if we do it like so:
```rust
let evals_to = EvaluatesTo::<Name>::new_unchecked(
        Value::Bool(true),
);

let serialized = serde_json::to_string(&evals_to).unwrap();

// Or Result<EvaluatesTo<Name>, _>
let deserialized: Result<EvaluatesTo<Value>, _> = serde_json::from_str(&serialized);
```

Reply:
> But it will if we do it like so:
> 
> ```rust
> let evals_to = EvaluatesTo::<Name>::new_unchecked(
>         Value::Bool(true),
> );
> 
> let serialized = serde_json::to_string(&evals_to).unwrap();
> 
> // Or Result<EvaluatesTo<Name>, _>
> let deserialized: Result<EvaluatesTo<Value>, _> = serde_json::from_str(&serialized);
> ```

this is what we need to prevent from happening